### PR TITLE
Small changes to allow Istiod to better run on knative env

### PR DESCRIPTION
**Pull request by costinm**
Change-Id: I69f1233627d555f104359a2fc1b152411f6b975c

Istiod can be run not only on K8S regular bods, but also on bare VMs or on KNative environment,
including managed versions using external load balancers. 

This PR provides better support - tested with GCP version of KNative, but should work with any other. 
In particular:
- detect it is running in knative and deal with a peculiarity of the load balancing/scaling, which expects first request to finish fast, temp until KNative provides a better way
-  allow using a different token type - Bearer is assumed to be 'platform token' ( at least on GCP)
- authenticate plain text - KNative takes care of TLS

All changes are off by default, protected with env that will only be set when running in the given envs.

Merged from a private branch, multiple contributors.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/xds/ads.go [hunk 1]
```diff
@@ -224,6 +231,14 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedD
 }
 
 func (s *DiscoveryServer) Stream(stream DiscoveryStream) error {
+	if knativeEnv != "" && firstRequest.Load() {
+		// How scaling works is the first request is the "loading" request. During
+		// loading request, concurrency=1. Once that request is done, we bump up to
+		// concurrency=80. But for us, the load request is 15min (based on timeout). As a
+		// result, we should exit the first request immediately.
```
Comment (howardjohn) [inline pilot/pkg/xds/ads.go:238]:
```suggestion
		// How scaling works in knative is the first request is the "loading" request. During
		// loading request, concurrency=1. Once that request is done, concurrency is enabled.
		// However, the XDS stream is long lived, so the first request would block all others. As a
		// result, we should exit the first request immediately; clients will retry.
```

File: pkg/kube/inject/webhook_test.go [hunk 2]
```diff
@@ -198,18 +198,6 @@ func TestInjectRequired(t *testing.T) {
 			},
 			want: false,
 		},
-		{
```
Comment (howardjohn) [inline pkg/kube/inject/webhook_test.go:201]: I don't think this is needed? not sure why the test should change

File: pkg/security/security.go [hunk 3]
```diff
@@ -58,6 +58,11 @@ var (
 	TokenAudiences = strings.Split(env.RegisterStringVar("TOKEN_AUDIENCES", "istio-ca",
 		"A list of comma separated audiences to check in the JWT token before issuing a certificate. "+
 			"The token is accepted if it matches with one of the audiences").Get(), ",")
+
+	XDSTokenType = env.RegisterStringVar("XDS_TOKEN_TYPE", "Bearer",
```
Comment (howardjohn) [inline pkg/security/security.go:62]: nit: this seems to be used for CA as well, but API has "XDS"

**Review by howardjohn — COMMENTED** [review]
looks good

**Review by howardjohn — APPROVED** [review]

File: pkg/security/security.go (hunk 3 above)
Comment (costinm) [reply pkg/security/security.go:62]: Ack, will do a follow-up - there are other related changes and infra that is using this.

**Pull request merged**